algosdk/atomic_transaction_composer.py

```diff
@@ -785,80 +803,78 @@ def execute(
+                tx_info={},
+                method=method,
+            )
+            try:
+                tx_info = cast(
+                    Dict[str, Any], client.pending_transaction_info(tx_id)
+                )
+                result = self.parse_result(
+                    method, self.tx_ids[method_index], tx_info
+                )
+            except Exception as e:
+                result.decode_error = e
+            method_results.append(result)
 
         return AtomicTransactionResponse(
             confirmed_round=confirmed_round,
             tx_ids=self.tx_ids,
             results=method_results,
         )
 
-    def parse_response(self, txns: List[Dict[str, Any]]) -> List[ABIResult]:
-        method_results = []
-        for i, tx_info in enumerate(txns):
-            tx_id = self.tx_ids[i]
-            raw_value: Optional[bytes] = None
-            return_value = None
-            decode_error = None
-
-            if i not in self.method_dict:
-                continue
-
-            # Parse log for ABI method return value
-            try:
-                if self.method_dict[i].returns.type == abi.Returns.VOID:
-                    method_results.append(
-                        ABIResult(
-                            tx_id=tx_id,
-                            raw_value=cast(bytes, raw_value),
-                            return_value=return_value,
-                            decode_error=decode_error,
-                            tx_info=tx_info,
-                            method=self.method_dict[i],
-                        )
-                    )
-                    continue
-
-                logs = tx_info["logs"] if "logs" in tx_info else []
-
-                # Look for the last returned value in the log
-                if not logs:
-                    raise error.AtomicTransactionComposerError(
-                        "app call transaction did not log a return value"
-                    )
-                result = logs[-1]
-                # Check that the first four bytes is the hash of "return"
-                result_bytes = base64.b64decode(result)
-                if (
-                    len(result_bytes) < 4
-                    or result_bytes[:4] != ABI_RETURN_HASH
-                ):
-                    raise error.AtomicTransactionComposerError(
-                        "app call transaction did not log a return value"
-                    )
-                raw_value = result_bytes[4:]
-                method_return_type = cast(
-                    abi.ABIType, self.method_dict[i].returns.type
-                )
-                return_value = method_return_type.decode(raw_value)
-            except Exception as e:
-                decode_error = e
-
-            method_results.append(
-                ABIResult(
+    def parse_result(
+        self, method: abi.Method, txid: str, txn: Dict[str, Any]
+    ) -> ABIResult:
+        tx_id = txid
+        raw_value = b""
```

Comment:
`bytes()`?

Reply:
they are functionally the same, but I changed all instances of empty byte literals to `bytes()` just in case it helps for readability in [5e2d2b7](https://github.com/algorand/py-algorand-sdk/pull/451/commits/5e2d2b70930d8fe2fc623becc941d743fa5fc1d0)